### DISCUSSION TITLE
Programmatically focus an input based input name

### DISCUSSION THREAD

**Discussion by samuelthng** [react-hook-form/react-hook-form#3761] [Q&A]
Is  there a way to focus any given input in the form based on the name, by using the same mechanism used to focus on errors?
Seems like a valid use case in some situations where you want to guide users to optional sections of the form using a button.

**Comment by jameswilson** [comment]
I have this same question. 🤚 

The recommended (React 16.8+ Hooks) way to handle focusing a random element when clicking on another element is via the `ref` attribute:

```
const FocusDemo = () => {

    const [inputRef, setInputFocus] = useFocus()

    return (
        <> 
            <button onClick={setInputFocus} >
               FOCUS
            </button>
            <input ref={inputRef} />
        </>
    )
    
}
```

```
const useFocus = () => {
    const htmlElRef = useRef(null)
    const setFocus = () => {htmlElRef.current &&  htmlElRef.current.focus()}

    return [ htmlElRef, setFocus ] 
}
```

However react-hook-form uses the `ref` field in another way.

So I'm not sure how to focus a random element, that is not hacky.  By hacky, I mean using the clicked button (or whatever it is) to navigate the DOM to find the element or using document.getElementByID("").

**Reply by bluebill1049** [reply]
we will introduce focusName for useFieldArray in v7, which you can focus any. also in v7 resolver, you will get hold of validating fields' ref, so you can focus them manually as well.

**Reply by jameswilson** [reply]
Interesting. Thank you for the response @bluebill1049.  Is there any suggestion on ~how~ a best-practice way to do this today?  and/or release roadmap for v7?

**Reply by bluebill1049** [reply]
Today you will need to host your own ref collection. V7: you can track down some progress here: https://github.com/react-hook-form/react-hook-form/discussions/3714 and https://github.com/react-hook-form/react-hook-form/pull/3741 working on this on my free with limited resource, so I can't give an accurate estimate.
